src/vm/hints/blake2s_utils.rs

```diff
@@ -144,6 +152,110 @@ pub fn finalize_blake2s(
     Ok(())
 }
 
+/* Implements Hint:
+    B = 32
+    MASK = 2 ** 32 - 1
+    segments.write_arg(ids.data, [(ids.low >> (B * i)) & MASK for i in range(4)])
+    segments.write_arg(ids.data + 4, [(ids.high >> (B * i)) & MASK for i in range(4)])
+*/
+pub fn blake2s_add_uint256(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    //Get variables from ids
+    let data_ptr = get_ptr_from_var_name("data", ids, vm, hint_ap_tracking)?;
+    let low_addr = get_relocatable_from_var_name("low", ids, vm, hint_ap_tracking)?;
+    let high_addr = get_relocatable_from_var_name("high", ids, vm, hint_ap_tracking)?;
+    let low = &vm.memory.get_integer(&low_addr)?.clone();
+    let high = &vm.memory.get_integer(&high_addr)?.clone();
+    //Main logic
+    //Declare constant
+    const MASK: u32 = u32::MAX;
+    const B: u32 = 32;
+    //Build first batch of data
+    let mut inner_data = Vec::<BigInt>::new();
+    for i in 0..4 {
+        inner_data.push((low >> (B * i)) & bigint_u32!(MASK));
```

Comment:
Let's create the mask only once at the beginning.

Reply:
Fixed!